gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java

```diff
@@ -469,6 +469,15 @@ public HttpJsonCallContext withDeadline(Instant newDeadline) {
     return withCallOptions(builder.setDeadline(newDeadline).build());
   }
 
+  public HttpJsonCallContext withRequestParamsDynamicHeaderOption(String requestParams) {
```

Comment:
We already have a `withExtraHeaders` method above which seems a better fit, this also makes us not have to add an extra field to `HttpJsonCallOptions`

Reply:
Ah I see. Let me try and use that.